### ISSUE TITLE
[BUG][DART] Multiple security options are all being added instead of only one.

### ISSUE THREAD

**Issue by gmeral** [OpenAPITools/openapi-generator#7057]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] What's the version of OpenAPI Generator used?
- [x] Have you search for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description

When multiple security options are listed, the openapi specs specifies that they should be options : 

_A declaration of which security mechanisms can be used for this operation. The list of values includes alternative security requirement objects that can be used. Only one of the security requirement objects need to be satisfied to authorize a request._

However the generated dart client function that adds security headers will attempt to add all of them instead of only one (see sample code below)

##### openapi-generator version

v 4.3.1

##### OpenAPI declaration file content or url

Here is the generated function responsible : 
```dart
void _updateParamsForAuth(List<String> authNames, List<QueryParam> queryParams, Map<String, String> headerParams) {
    authNames.forEach((authName) {
      Authentication auth = _authentications[authName];
      if (auth == null) throw ArgumentError("Authentication undefined: " + authName);
      auth.applyToParams(queryParams, headerParams);
    });
  }
```

Here is the security object of the scheme in the spec
```json
"security": [
          {
            "ApiKeyAuth": []
          },
          {
            "BearerAuth": []
          }
        ]
```

##### Command line used for generation

openapi-generator generate -v -i spec.json -g dart -o api-client/ --api-name-suffix myproject -c config.json

##### Steps to reproduce

1. generate a dart client including  a schema with multiple security options.
2. try to make a related call with only one security option set (e.g only ApiKeyAuth)
3. client throws an exception because "BearerAuth" is undefined

##### Suggest a fix

Provide an argument to select a preferred security options or when a valid authorization header is set, ignore the others.

**Comment by jld3103** [comment]
I'm also experiencing this problem and it makes APIs that have multiple authentication methods unusable. The authentication option handling in general could be improved a lot, because it's quite annoying to actually setup the authentication options.
I might make a PR to fix this and improve the authentication option handling in general.

@jaumard @josh-burton @amondnet @sbu-WBT @kuhnroyal @agilob @ahmednfwela

**Comment by agilob** [comment]
Is this still a problem on 6.x branch @jld3103 ?

**Comment by jld3103** [comment]
I'm using the master branch right now and the problem still exists there. I don't know about the 6.x branch, but I guess it's the same there.

**Comment by ahmednfwela** [comment]
> I'm using the master branch right now and the problem still exists there. I don't know about the 6.x branch, but I guess it's the same there.

6.x has breaking changes from master, so try it there as well

**Comment by jld3103** [comment]
Ok I'll take a look later.  
How does the development model work? Will the 6.x branch later be merged into the master branch? And is the master branch basically 5.x development?

**Comment by agilob** [comment]
6.x will be master branch after the next major release. major release (every 1 year) can contain breaking changes

**Comment by jld3103** [comment]
I checked and it's exactly the same problem.

**Comment by jld3103** [comment]
@agilob See my fix targeting the 6.0.x branch.
